woocommerce/woocommerce
# [CYS - E2E tests] Add E2E tests for the loading screen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds E2E tests to cover the loading screen. This PR ensures that:
- the loading screen shows the right labels and makes the network requests to set up the store.
- The loading screen will redirect you to the intro page in case of network errors.
- The loading screen is loaded only when the store hasn't been set up.


Closes #45343.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Run `pnpm run --filter="@woocommerce/plugin-woocommerce" test:e2e-pw plugins/woocommerce/tests/e2e-pw/tests/customize-store/loading-screen/loading-screen.spec.js` and ensure that everything is green



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

[CYS - E2E tests] Add E2E tests for the loading screen

</details>
